File: Programmers/짝지어제거하기.cpp
#include <iostream>
#include <string>
#include <stack>
using namespace std;

int solution(string s)
{
    if(s.length()%2==1) return 0;
    stack <char> st;
    
    for(int i=0; i<s.length(); i++){
        if(!st.empty() && st.top() == s[i]) st.pop();
        else st.push(s[i]);
    }

    if(st.empty()) return 1;
    else return 0;
}

File: JavaScript/Programmers/짝지어제거하기.js
function solution(s)
{
    if(s.length%2===1) return 0;
    
    let save = [];
    for(let i = 0; i<s.length; i++){
        if(save[save.length-1]!==s[i]||save.length===0) save.push(s[i]);
        else save.pop();
    }
  
    if(save.length===0) return 1;
    else return 0;
}